core/state_processor.go

```diff
@@ -95,6 +95,18 @@ func ApplyTransaction(config *params.ChainConfig, bc ChainContext, author *commo
 	// Create a new environment which holds all relevant information
 	// about the transaction and calling mechanisms.
 	vmenv := vm.NewEVM(context, statedb, config, cfg)
+
+	if config.IsYoloV2(header.Number) {
+		statedb.AddAddrToAccessList(msg.From())
+		if dst := msg.To(); dst != nil {
+			statedb.AddAddrToAccessList(*dst)
+			// If it's a create-tx, the destination will be added inside evm.create
+		}
+		for _, addr := range vmenv.ActivePrecompiles() {
```

Comment:
Shouldn't this rather be set up in ApplyMessage? Otherwise the adding to the access list has to be replicated in every test

Reply:
I'm not sure which is best. On one hand, it would be nice to move it even earlier, so the active precompiles are added only once per block. On the other hand, doing it later means we don't have to do it in several different places. 

One thing I just thought of, btw, is toe look into how the miner `currentState` works. The access list in the state must be cleared between transactions.

Reply:
Access List are different for each transaction, how can you add precompiles once per block?
Access List should be cleared after every transaction.

Reply:
No, the transaction list should not be deleted and compal should be done once, and each amount of ether should be placed in the person's software wallet

Reply:
@09306677806 am not sure if i got you exactly on this, what do you mean by transaction list?